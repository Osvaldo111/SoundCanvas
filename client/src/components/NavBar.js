import React from "react";
import "../style/navBar.css";
import { connect } from "react-redux";
import { isResetBttn } from "../actions";
import { isDownloadCanvas } from "../actions";
import { isSendCanvas } from "../actions";
import SendForm from "./SendForm";
/**
 * @author Osvaldo Carrillo
 * Date: 02/08/2020
 * This class is the top navigation bar
 * of the application.
 */
class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displaySendForm: false,
      disSendFormBttn: false,
      messageSentEmail: ""
    };
  }
  isResetBttnPress = () => {
    const { gfCompleted } = this.props.canvasProps;
    if (gfCompleted) {
      this.props.isResetBttn(true);
    } else {
      alert("Please, record something");
    }
  };

  isDownloadBttnPress = () => {
    const { gfCompleted } = this.props.canvasProps;
    if (gfCompleted) {
      this.props.isDownloadCanvas(true);
    } else {
      alert("Please, record something");
    }
  };

  isSendBttnPress = () => {
    const { gfCompleted } = this.props.canvasProps;
    if (gfCompleted) {
      this.props.isSendCanvas(true);
    } else {
      alert("Please, record something");
    }
  };

  render() {
    return (
      <div className="navBar">
        <div className="navBarLogo">
          <p>Sound Canvas</p>
        </div>
        <div className="navBarBttn">
          <button onClick={this.isResetBttnPress} id="resetBttn">
            <p>Reset</p>
          </button>
          <a
            className="bttnEffect"
            id="download"
            download="myImage.jpg"
            onClick={this.isDownloadBttnPress}
          >
            <div id="spinAnim"></div>
            <p>Download</p>
          </a>
          <a
            className="bttnEffect bttnSend"
            id="download"
            download="myImage.jpg"
            onClick={this.isSendBttnPress}
          >
            <div id="spinAnim"></div>
            <p>Send</p>
          </a>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    canvasProps: state.canvas
  };
}
const mapDispatchToProps = {
  isResetBttn,
  isDownloadCanvas,
  isSendCanvas
};
export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
