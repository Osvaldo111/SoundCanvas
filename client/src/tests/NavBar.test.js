import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NavBar from "../components/NavBar";
import { mount, shallow } from "enzyme";
//Redux
import { Provider } from "react-redux";
import store from "../reducers/store";
import { createSerializer } from "enzyme-to-json";
import { spy } from "sinon";
configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Navigation bar <NavBar />", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <NavBar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("check when the user press the reset button", () => {
    window.alert = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <NavBar />
      </Provider>
    );
    // console.log(wrapper.find("NavBar").instance().props, "************");
    const navBar = wrapper.find("NavBar").instance();
    const navSpy = spy(navBar, "isResetBttnPress");
    navBar.forceUpdate();
    wrapper.find("#resetBttn").simulate("click");
    expect(navSpy.calledOnce).toEqual(true);
  });
});
