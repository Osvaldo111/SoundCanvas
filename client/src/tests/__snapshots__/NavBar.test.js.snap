// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar <NavBar /> renders correctly 1`] = `
<Connect(NavBar)>
  <div
    className="navBar"
  >
    <div
      className="navBarLogo"
    >
      <p>
        Sound Canvas
      </p>
    </div>
    <div
      className="navBarBttn"
    >
      <button
        id="resetBttn"
        onClick={[Function]}
      >
        <p>
          Reset
        </p>
      </button>
      <a
        className="bttnEffect"
        download="myImage.jpg"
        id="download"
        onClick={[Function]}
      >
        <div
          id="spinAnim"
        />
        <p>
          Download
        </p>
      </a>
      <a
        className="bttnEffect bttnSend"
        download="myImage.jpg"
        id="download"
        onClick={[Function]}
      >
        <div
          id="spinAnim"
        />
        <p>
          Send
        </p>
      </a>
    </div>
  </div>
</Connect(NavBar)>
`;
